ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-ref.yaml

```diff
@@ -14,5 +14,7 @@ ref:
   dependencies:
   - name: "release:latest"
     env: OPENSHIFT_INSTALL_RELEASE_IMAGE
+  - name: "release:latest"
+    env: OPENSHIFT_UPGRADE_RELEASE_IMAGE
```

Comment:
Any specific reason to define a new distinct var? I guess we could reuse https://github.com/openshift/release/blob/b817deec43a3ed9f5d76348ef5cfd92ed13de89c/ci-operator/step-registry/baremetalds/e2e/upgrade/baremetalds-e2e-upgrade-workflow.yaml#L12

Reply:
I don't think `OVERRIDE` is all that appropriate here.  I felt strongly enough to break from tradition with my new variable, but not strongly enough to try and update existing variables.

Reply:
One day I'll get those steps moved too by reviving #12959.  I just haven't been able to track down approvers yet.  Anyhow, having the scripts prefer one and fall back to another is pretty easy:

```sh
OPENSHIFT_UPGRADE_RELEASE_IMAGE="${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE:-${OPENSHIFT_UPGRADE_RELEASE_IMAGE}}"
some_consumer "${OPENSHIFT_UPGRADE_RELEASE_IMAGE}"
```

But also, this is not my step, so feel free to file alternative PRs with whatever handling makes the most sense to you; no need to just follow my preferences ;)